test/functional/launch/nested_tile_direct/tests/test-launch-nested-Tile-Direct.hpp

```diff
@@ -0,0 +1,182 @@
+
+  INDEX_TYPE N = static_cast<INDEX_TYPE>(RAJA::stripIndexType(N1) *
+                                         RAJA::stripIndexType(N2) *
+                                         RAJA::stripIndexType(N3));
+
+  camp::resources::Resource working_res{WORKING_RES::get_default()};
+  INDEX_TYPE* working_array;
+  INDEX_TYPE* check_array;
+  INDEX_TYPE* test_array;
+
+  size_t data_len = RAJA::stripIndexType(N);
+  if ( data_len == 0 ) {
+    data_len = 1;
+  }
+
+  allocateForallTestData<INDEX_TYPE>(data_len,
+                                     working_res,
+                                     &working_array,
+                                     &check_array,
+                                     &test_array);
+
+  if ( RAJA::stripIndexType(N) > 0 ) {
+
+    std::iota(test_array, test_array + RAJA::stripIndexType(N), 0);
+
+    constexpr int DIM = 3;
+    using layout_t = RAJA::Layout<DIM, INDEX_TYPE,DIM-1>;
+    RAJA::View<INDEX_TYPE, layout_t> Aview(working_array, N3, N2, N1);
+
+    RAJA::launch<LAUNCH_POLICY>
+      (RAJA::LaunchParams(RAJA::Teams(blocks_x, blocks_y, blocks_z), RAJA::Threads(threads_x, threads_y,threads_z)),
+        [=] RAJA_HOST_DEVICE(RAJA::LaunchContext ctx) {
+
+        RAJA::tile<TEAM_Z_POLICY>(ctx, tile_size_z, r3, [&](RAJA::TypedRangeSegment<INDEX_TYPE> const &z_tile) {
+            RAJA::tile<TEAM_Y_POLICY>(ctx, tile_size_y, r2, [&](RAJA::TypedRangeSegment<INDEX_TYPE> const &y_tile) {
+                RAJA::tile<TEAM_X_POLICY>(ctx, tile_size_x, r1, [&](RAJA::TypedRangeSegment<INDEX_TYPE> const &x_tile) {
+
+                    RAJA::loop<THREAD_Z_POLICY>(ctx, z_tile, [&](INDEX_TYPE tz) {
+                        RAJA::loop<THREAD_Y_POLICY>(ctx, y_tile, [&](INDEX_TYPE ty) {
+                            RAJA::loop<THREAD_X_POLICY>(ctx, x_tile, [&](INDEX_TYPE tx) {
+
+                                auto idx = tx + N1 * (ty + N2 * tz);
+
+                                Aview(tz, ty, tx) = static_cast<INDEX_TYPE>(idx);
+
+                              });
+                          });
+                      });
+
+                  });
+              });
+          });
+    });
+  } else { // zero-length segment
+
+    memset(static_cast<void*>(test_array), 0, sizeof(INDEX_TYPE) * data_len);
+
+    working_res.memcpy(working_array, test_array, sizeof(INDEX_TYPE) * data_len);
+
+    RAJA::launch<LAUNCH_POLICY>
+      (RAJA::LaunchParams(RAJA::Teams(blocks_x, blocks_y, blocks_z), RAJA::Threads(blocks_x, blocks_y ,blocks_z)),
+        [=] RAJA_HOST_DEVICE(RAJA::LaunchContext ctx) {
+
+        RAJA::tile<TEAM_Z_POLICY>(ctx, threads_z, r3, [&](RAJA::TypedRangeSegment<INDEX_TYPE> const &z_tile) {
+            RAJA::tile<TEAM_Y_POLICY>(ctx, threads_y, r2, [&](RAJA::TypedRangeSegment<INDEX_TYPE> const &y_tile) {
+                RAJA::tile<TEAM_X_POLICY>(ctx, threads_x, r1, [&](RAJA::TypedRangeSegment<INDEX_TYPE> const &x_tile) {
+
+                    RAJA::loop<THREAD_Z_POLICY>(ctx, z_tile, [&](INDEX_TYPE tz) {
```

Comment:
Use this macro instead? https://github.com/LLNL/RAJA/blob/develop/include/RAJA/util/macros.hpp#L67